### PR TITLE
Refactor PAI/Alisa submitter

### PR DESCRIPTION
**Pull request by Yancey0623**
This PR refactor `PAISubmitter` and `AlisaSubmitter` since they have the common logic: package Tensorflow code and submit job with PAI cmd, will do as the following:
1. Rename `PAISubmitter` to `MaxComputeSubmitter` as it's submitted a job using `odpscmd` instead of `PAI`.
1. `pai_codegen` generates the PAI-Tensorflow program and PAI program.
1. `maxcomputeSubmitter` submit the job using `odpscmd`.
1. `alisaSubmitter` submit the job using `goalisa`.

### CODE REVIEW

File: pkg/sql/alisa_submitter.go [hunk 1]
```diff
@@ -137,10 +120,12 @@ func (s *alisaSubmitter) cleanUpModel(modelPath string) error {
 	if len(sub) != 3 {
 		return fmt.Errorf("SQLFLOW_OSS_CHECKPOINT_DIR should be format: oss://bucket/?role_arn=xxx&host=xxx")
 	}
-	bucket, e := getBucket(sub[2], os.Getenv("SQLFLOW_OSS_AK"), os.Getenv("SQLFLOW_OSS_SK"), sub[1])
+	fmt.Println(os.Getenv("SQLFLOW_OSS_MODEL_ENDPOINT"), os.Getenv("SQLFLOW_OSS_AK"), os.Getenv("SQLFLOW_OSS_SK"), sub[1])
```
Comment (wangkuiyi) [inline pkg/sql/alisa_submitter.go]: Is this line printing a log message? Should we call `log.Print` or just remove it?

File: pkg/sql/alisa_submitter.go [hunk 2]
```diff
@@ -61,74 +60,58 @@ func (s *alisaSubmitter) submitAlisaTask(code, resourceName string) error {
...
 		return e
 	}
+	return s.alisaCmd(code, paiCmd, ossObjectName)
+}
 
-	code, e := pai.TFTrainAndSave(ts, s.Session, modelPath, cc)
-	if e != nil {
-		return e
-	}
+func (s *alisaSubmitter) alisaCmd(program, alisaCode, ossObjectName string) error {
```
Comment (typhoonzero) [inline pkg/sql/alisa_submitter.go]: `alisaCmd` does uploading code resource and run the submit. How about splitting these two operations to two functions?

File: pkg/sql/alisa_submitter.go [hunk 3]
```diff
@@ -137,10 +120,12 @@ func (s *alisaSubmitter) cleanUpModel(modelPath string) error {
 	if len(sub) != 3 {
 		return fmt.Errorf("SQLFLOW_OSS_CHECKPOINT_DIR should be format: oss://bucket/?role_arn=xxx&host=xxx")
 	}
-	bucket, e := getBucket(sub[2], os.Getenv("SQLFLOW_OSS_AK"), os.Getenv("SQLFLOW_OSS_SK"), sub[1])
+	fmt.Println(os.Getenv("SQLFLOW_OSS_MODEL_ENDPOINT"), os.Getenv("SQLFLOW_OSS_AK"), os.Getenv("SQLFLOW_OSS_SK"), sub[1])
+	bucket, e := getBucket(os.Getenv("SQLFLOW_OSS_MODEL_ENDPOINT"), os.Getenv("SQLFLOW_OSS_AK"), os.Getenv("SQLFLOW_OSS_SK"), sub[1])
 	if e != nil {
 		return e
 	}
+	fmt.Println("delete oss", modelPath)
```
Comment (typhoonzero) [inline pkg/sql/alisa_submitter.go]: Change to `log.Print` or remove this

File: pkg/sql/maxcompute_submitter.go [hunk 4]
```diff
@@ -28,7 +30,9 @@ import (
 	"sqlflow.org/sqlflow/pkg/sql/codegen/pai"
 )
 
-type paiSubmitter struct{ *defaultSubmitter }
+var tarball = "job.tar.gz"
```
Comment (typhoonzero) [inline pkg/sql/maxcompute_submitter.go]: Why rename `pai_submitter` to `maxcompute_submitter`? I think we should keep the name `maxcompute` always stands for a database driver. And, PAI as one of the execution platform.

Comment (Yancey0623) [reply pkg/sql/maxcompute_submitter.go]: It's just because of submitting a PAI job using `odpscmd` instead of the PAI command.

Comment (Yancey0623) [reply pkg/sql/maxcompute_submitter.go]: I revert the renaming, can renaming it in the future if we want to submit PAI job in varients ways as as PAI command line.

File: pkg/sql/alisa_submitter.go (hunk 2 above)
Comment (Yancey0623) [reply pkg/sql/alisa_submitter.go]: I renamed it to `uploadResourceAndSubmitAlisaTask`, maybe it's a common function can be used in `ExecuteTrain` , `ExecutePredict` and `ExecuteExplain`

File: pkg/sql/alisa_submitter.go (hunk 1 above)
Comment (Yancey0623) [reply pkg/sql/alisa_submitter.go]: Just used for debugging, I removed it.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**